cfme/tests/automate/custom_button/test_infra_objects.py

```diff
@@ -0,0 +1,87 @@
+import pytest
+import fauxfactory
+
+from widgetastic_patternfly import Dropdown
+
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.markers.env_markers.provider import ONE_PER_TYPE
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.log import logger
+
+
+pytestmark = [
+    pytest.mark.tier(2),
+    pytest.mark.usefixtures("setup_provider"),
+    pytest.mark.provider([VMwareProvider], selector=ONE_PER_TYPE),
+]
+
+INFRA_OBJECTS = ["PROVIDER"]
+
+DISPLAY_NAV = {
+    "Single entity": ["Details"],
+    "List": ["All"],
+    "Single and list": ["All", "Details"],
+}
+
+
+@pytest.fixture(
+    params=INFRA_OBJECTS, ids=[obj.capitalize() for obj in INFRA_OBJECTS], scope="module"
+)
+def button_group(appliance, request):
+    collection = appliance.collections.button_groups
+    button_gp = collection.create(
+        text=fauxfactory.gen_alphanumeric(),
+        hover=fauxfactory.gen_alphanumeric(),
+        type=getattr(collection, request.param),
+    )
+    yield button_gp, request.param
```

Comment:
Question : Why do we need request.param ?
I don't see it being used in test

Reply:
Nice catch but,
As an initial comment only. I mention that its structural PR for custom button parametrize automation.
At the initial level, I committed with `provider` custom button type object but other objects are going to add in `INFRA_OBJECTS` list. I need this for uncollecting object types as per appliance version support. 
for example:
```
@pytest.mark.uncollectif(lambda button_group: "PROVIDER" in button_group)
```

Please, for now, you can ignore this. Its usable thing in next PR only.